---
title: "ChIP_Wt1 - Results"
author: "Kelterborn"
date: "`r Sys.Date()`"
output:
  md_document:
    variant: gfm
always_allow_html: true
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      results = "hide", error=FALSE, warning=FALSE, message=FALSE)
options(kableExtra.auto_format = FALSE)
```

# R Prepare System
## R update and load librarys
BiocManager::install()
BiocManager::install("plyranges")

```{r librarys}
# BiocManager::install(update = TRUE, ask = FALSE)

library(dbplyr)
library(tidyverse)
library(ChIPseeker)
library(rtracklayer)
library(trackViewer)
library(GenomicRanges)
library(IRanges)
library(ChIPpeakAnno)
library(AnnotationHub)
library(ggplot2)
library(viridis)
library(kableExtra)
library(DT)
library(patchwork)
library(gridExtra)
library(foreach)
library(doParallel)
library(tidyGenomeBrowser)
library(Gviz)
library(magrittr)
library(plyranges)

# library()

library(TxDb.Mmusculus.UCSC.mm39.knownGene)
txdb <- TxDb.Mmusculus.UCSC.mm39.knownGene
library(org.Mm.eg.db)

```

```{r old_librarys, eval=FALSE, include=FALSE}
# library("ggVennDiagram")
# library("ggvenn")
# 

# library(ChIPQC)
# library(BiocParallel)
# library(readxl)
# library(writexl)
# library(xlsx)
# library(beepr)

# library(TxDb.Hsapiens.UCSC.hg19.knownGene)
# library(TxDb.Hsapiens.UCSC.hg38.knownGene)
# library(TxDb.Mmusculus.UCSC.mm10.knownGene)
# library(TxDb.Mmusculus.UCSC.mm39.knownGene)
# # txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
# # txdb <- TxDb.Mmusculus.UCSC.mm10.knownGene
# txdb <- TxDb.Mmusculus.UCSC.mm39.knownGene
# library(clusterProfiler)
# library(UpSetR)
# library(ggimage)
# library(ReactomePA)
# library(ggcorrplot)
# library(gplots)
# library(pheatmap)
# library(EnsDb.Mmusculus.v79)
# library(Rsamtools)
# library(VennDiagram)
# library(biomaRt)
# library(biobtreeR)
# library(memes)
# library(magrittr)
# 

# library(Motif2Site)
# library(updater)
# 
# library(BiocManager)
# library(ChIPQC)
# library(rtracklayer)

# 
# 

# library(GenomicFeatures)
# library(UpSetR)
# library(grid)
# library(plyr)

# library(rentrez)
# library("GenomicFeatures")
# library("Gviz")
# library(GenomicRanges)
# library()


```


## R folders
```{r R_folders, warning=FALSE, echo=F}

if(Sys.info()[["sysname"]]== "Linux") {
  s <- "/mnt/s"} else(
  s <- "S:")

ddir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF",sep="/")
gitdir <- paste(ddir,"ChIPseq_Wt1_P3026", sep="/")
resdir <- paste(gitdir,"Results", sep="/")
pubfigdir <- paste(resdir,"pub_figures",sep="/")

wdir <- "E:/Simon/P3026_ChIP_Seq_epiSVF"

# setwd(gitdir)
getwd()


```

# Unix Prepare System
## Unix Update System
```{bash unix_system, eval=FALSE, echo=F}
##  linux ####
sudo apt update
sudo apt upgrade
conda update conda
conda update --all
sudo mount -t drvfs S: /mnt/s

```

## Unix Folder & Names
```{bash unix_folders, message=FALSE, warning=FALSE, echo=FALSE}
source activate chipseq

run="Kelly";

dirdata="/mnt/s/AG/AG-Scholz-NGS/Daten/ChIP_epiSVF_P3026";
dir="/mnt/e/Simon/P3026_ChIP_Seq_epiSVF"; 
dirs="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF"; 

index_GRCh38="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Human/GRCh38_noalt_decoy_as/GRCh38_noalt_decoy_as";
index_GRCm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/GRCm39";
index_mm10="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/mm10/mm10";
index_mm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/USCS/mm39";
index_cm_mm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/USCS/mm39.index";

fastqdir="$dirdata/fastq";

ardir="$dirdata/Adapterremoval";
  test -d $ardir && echo "# Folder $ardir exists" ||
  (mkdir $ardir && echo "# Folder $ardir created");
  
fastqcdir="$dirdata/fastqc/";
  test -d $fastqcdir && echo "# Folder $fastqcdir exists" ||
  (mkdir $fastqcdir && echo "# Folder $fastqcdir created");
  
multiqcdir="$dirdata/multiqc/";
  test -d $multiqcdir && echo "# Folder $multiqcdir exists" ||
  (mkdir $multiqcdir && echo "# Folder $multiqcdir created");
  
bdir="$dir/Bowtie2";
  test -d $bdir && echo "# Folder $bdir exists" ||
  (mkdir $bdir && echo "# Folder $bdir created");
  
mdir="$dirs/MACS3";
  test -d $mdir && echo "# Folder $mdir exists" ||
  (mkdir $mdir && echo "# Folder $mdir created");

rgtmdir="$dir/RGT-Motif";
  test -d $rgtmdir && echo "# Folder $rgtmdir exists" ||
  (mkdir $rgtmdir && echo "# Folder $rgtmdir created");

dtoolsdir="$dir/deeptools";
  test -d $dtoolsdir && echo "# Folder $dtoolsdir exists" ||
  (mkdir $dtoolsdir && echo "# Folder $dtoolsdir created");

chromapdir="$dir/chromap";
  test -d $chromapdir && echo "# Folder $chromapdir exists" ||
  (mkdir $chromapdir && echo "# Folder $chromapdir created");

blacklhs="/mnt/e/Simon/ChIPSeq_Kelly/Bowtie2/hg38-blacklist.v2.bed"
blacklmm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/mm39.excluderanges.bed.gz"
blacklmm10="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/mm10/mm10.blacklist.bed.gz"

```

# 3 Results

[Analysis Overview](../Data/sheme.pdf){height=100%, width=100%}


```{r method_overview2, echo = TRUE, include=TRUE, results="show"}
# write outside r code: ![Analysis Overview](../Data/sheme.png){height=100%, width=100%}

# knitr::include_graphics("https://github.com/DNAborn/ChIPseq_Wt1/blob/main/Results/sheme.png")
# knitr::include_graphics("https://github.com/DNAborn/ChIPseq_Wt1/blob/main/Results/sheme.png?raw=true")
knitr::include_graphics("../Data/sheme.png")

# knitr::include_graphics("./sheme.png")
# knitr::include_graphics("./sheme.pdf")
# knitr::include_graphics("../sheme.png")
# knitr::include_graphics("../sheme.pdf")
# knitr::include_graphics("/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/ChIPseq_Wt1_P3026/Results/sheme.png")
# knitr::include_graphics("/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/ChIPseq_Wt1_P3026/Results/sheme.pdf")
getwd()
```


```{r Method_overview, message=TRUE, fig.align='center', fig.cap='Analysis Overview', out.width='1\\linewidth', fig.pos='H'}
print("include figure in r chunk")
knitr::include_graphics("../Data/sheme.pdf") # doesn't show

```


### Generate combined peak list
```{r combine_peaks, echo=FALSE, }
csdir <- paste(ddir,"ChIPSeeker",sep="/")

# Peak Calling files
macsdir <- paste(ddir,"MACS3",sep="/")
macssubdir <- list.dirs(macsdir)
macssubdir <- macssubdir[c(6,5,3,2,4)]

# blacklist
blacklmm39 <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/mm39.excluderanges.bed.gz",sep="/")
bl <- readPeakFile(blacklmm39)

extraCols_narrowPeak <- c(signalValue = "numeric", pValue = "numeric",
                          qValue = "numeric", peak = "integer")

# Read peak files from all subfolders
# generate list with combined peaks
npeak_list_all <- list()
npeak_combined_all <- list()
npeak_numbers_table <- {}
hist_hits <- list()

# loop through subfolders
for (beddir in macssubdir){
  print(beddir)
  run <- basename(beddir)
  cat("##########\n#",run)

beds <- list.files(beddir, pattern=".narrowPeak")
beds <- file.path(beddir,beds)
paste("Number of replicates:",length(beds))
name <- paste(run,"_",sep="")

npeak_list <- list()
i <- beds[8]
# inside loop thourgh each replicate
for (i in beds) {
  npeak_numbers <- {}
  # print(i)
  # npeak <- readPeakFile(i)
npeak <- import(i, format = "BED",extraCols = extraCols_narrowPeak)
n <- basename(i) %>% str_remove(pattern="_all_peaks.narrowPeak")
print(paste("# Run: ",run,", Sample: ",n," Peaks: ",length(npeak), sep=""))
npeak_numbers <- rbind(npeak_numbers,data.frame(sample = paste(n,sep="_"), Mapping = run, all_peaks = length(npeak)))
assign(paste(run,"npeaks",sep="_"),npeak_numbers)

print(c("## Total peaks",length(npeak)))
npeak <- keepStandardChromosomes(npeak, pruning.mode="coarse")
print(c("## Peaks on Std Chromosomes",length(npeak)))
npeak_numbers <- cbind(npeak_numbers, filt_Std_Chrom = length(npeak))
filtered2 <- findOverlaps(npeak,bl, type = 'any')
queryHits(filtered2) %>% duplicated() %>% summary
filtered <- npeak[-(queryHits(filtered2)),]
revmap <- findOverlaps(filtered, npeak, select="arbitrary")
print(c("## Peaks filtered from blacklist:",length(npeak)-length(filtered),"Remaining:",length(filtered)))
npeak_numbers <- cbind(npeak_numbers, filt_bl = length(filtered))
npeak_list[[n]] <- filtered
npeak_list_all[[n]] <- filtered
npeak_numbers_table <- rbind(npeak_numbers_table,npeak_numbers)
}

# combine replicates
assign(paste(run,"npeak_list",sep="_"),npeak_list)
npeak_list <- GRangesList(npeak_list)
ul <- unlist(npeak_list)
ul <- sort(ul,by=~qValue, decreasing = T)
length(ul)
npeak_combined <- GenomicRanges::reduce(ul)
length(ul)
length(npeak_combined)
revmap <- findOverlaps(npeak_combined, ul, select="first")

# Means of mcols...
revmap_h <- findOverlaps(npeak_combined, ul, select="all") # "first"


# set-up parallel
n.cores <- parallel::detectCores()/2
my.cluster <- parallel::makeCluster(
  n.cores, 
  type = "FORK")
print(my.cluster)
doParallel::registerDoParallel(cl = my.cluster)
foreach::getDoParRegistered()
foreach::getDoParWorkers()

# parallel loop
i <- 5549
stats <- {}
stats_t <- foreach(
  i = 1:length(npeak_combined), 
  .combine = 'rbind'
) %dopar% {
  op <- revmap_h[queryHits(revmap_h)==i] %>% to()
  op_df <- mcols(ul)[op,] %>% as.data.frame()
  max.row <- which.max(op_df$score)
  stats <- op_df[max.row,]
  # stats <- summarize_all(op_df[,2:4], max)
  stats$id <- i
  stats$name_list <- paste(op_df$name, collapse =", ")
  o <- subset(ul,name==stats$name)
  max.center <- start(o)+o$peak
  stats$max.center <- max.center
  stats$max.start <- start(o)
  stats$max.end <- end(o)
  stats$max.width <- width(o)
  stats
}
parallel::stopCluster(cl = my.cluster)

head(stats_t)
mcols(npeak_combined) <- stats_t


i <- 2

# # standard loop
# for (i in 1:10000){
# op <- revmap_h[queryHits(revmap_h)==i] %>% to()
# op_df <- mcols(ul)[op,] %>% as.data.frame()
# Stats <- summarize_all(op_df, mean)
# Stats$name <- paste(op_df$name, collapse =", ")
# Stats$id <- i
# Stats_t <- rbind(Stats_t,Stats)
# print(i)
# }
# Stats_t
# mcols(npeak_combined) <- mcols(ul)[revmap, ,drop=FALSE]
length(npeak_combined)

# count overlap between peaks
npeak_combined$hits <- rep(0,length(npeak_combined))
for (i in 1:length(npeak_list)){
# print(i)
npeak_combined$hits <- npeak_combined$hits+countOverlaps(npeak_combined,npeak_list[i])
}
# hist(c(0,10,npeak_combined$hits), xlim=c(0,10), breaks=10, main=c("hits per replicate, Method: ",run))

# generate list of histograms
gg_b <- ggplot_build(
  ggplot(as.data.frame(npeak_combined),aes(x=hits, colours=hits)) +
    stat_bin(alpha=0.5, position = 'identity', binwidth = 1, colours = npeak_combined$hits) + 
    scale_fill_viridis_c() +
    coord_cartesian(xlim = c(1, 9)) +
    stat_bin(position = 'identity', binwidth=1, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
    scale_x_continuous(n.breaks= 9) +
    ggtitle(paste("peaks per replicate. Total peaks (>1 hit):"))
)
nu_bins <- dim(gg_b$data[[1]])[1]
ymax <- max(gg_b$data[[1]]$y)
g <- ggplot(as.data.frame(npeak_combined),aes(x=hits, colours=hits)) +
    stat_bin(alpha=0.5, position = 'identity', binwidth = 1, colours = npeak_combined$hits, fill = viridis(nu_bins)) + 
    scale_fill_viridis_c() +
    coord_cartesian(xlim = c(1, 9), ylim=c(0,ymax*1.05)) +
    stat_bin(position = 'identity', binwidth=1, geom='text', size=3,  vjust=-0.2, aes(label=after_stat(count)), colour=c("red",rep("blue",nu_bins-1))) +
    scale_x_continuous(n.breaks= 9) +
    ggtitle(paste("A) Method:",run,"Peaks per replicate. Total peaks (>1 hit):",summary(elementMetadata(npeak_combined)$hits > 1)["TRUE"]))
# 
print(paste("# finished",run,"- generate ggplot"))
hist_hits[[run]] <- g

summary(elementMetadata(npeak_combined)$hits > 1)[2]
# npeak_combined <- npeak_combined[elementMetadata(npeak_combined)$hits > 1,]
length(npeak_combined)
assign(paste(run,"npeak_combined",sep="_"),npeak_combined)

npeak_combined_all[[run]] <- npeak_combined

}

# npeak_combined_all
# lapply(npeak_combined_all, length)

# generate overlap table
peaks_overlap <- data.frame(Sample=basename(macssubdir),
                          Peaks=lengths(npeak_combined_all))
ol.all <- {}
l <- length(npeak_combined_all)
for (n in names(npeak_combined_all)){
  assign(paste("ol",n,sep="."),{})
  variable <- paste0("ol.",n)
  ol <- eval(parse(text = variable))
  peaks_to_compare <- npeak_combined_all[[n]]
for (i in 1:l){
  start(peaks_to_compare) <- start(peaks_to_compare) - 100
  end(peaks_to_compare) <- end(peaks_to_compare) + 100
  ol <- c(ol,sum(countOverlaps(peaks_to_compare,npeak_combined_all[[i]])))
}
  assign(paste("ol",n,sep="."),ol)
}

peaks_overlap <- data.frame(peaks_overlap,ol.Std, ol.pe, ol.local, ol.chro5k, ol.Online)
# peaks_overlap

```

#### Peak Tables
```{r peak_numbers, include=TRUE, echo = TRUE, results='markup'}
dim(npeak_numbers_table)
knitr::kable(npeak_numbers_table, format = "markdown") %>% 
  kable_styling("striped", full_width = F) %>% 
  scroll_box(height = "400px")

# datatable(npeak_numbers_table)

knitr::kable(peaks_overlap)  %>% 
  kable_styling("striped", full_width = F) %>% 
  scroll_box(height = "400px")

```


#### Hists & Tables
```{r hits_tables, echo=FALSE, results = "show", fig.width=12, fig.height=8, out.width="100%"}
# fig.height=10, fig.width=10, out.height="2000px", 

# patchwork::wrap_plots(hist_hits) +plot_layout(nrow = 3, ncol = 2, axis_titles = "collect") # axes = "collect"

# add sum row to tables
ht_list <- list()
for (run in names(hist_hits)){
     table <- subset(npeak_numbers_table, Mapping == run) %>% 
                bind_rows(summarise_all(., ~if(is.numeric(.)) sum(.) else "Total"))
      l <- dim(table)[1]
      t1 <- ttheme_default(base_size = 7,
                         core=list(
        fg_params=list(fontface=c(rep("plain", (l-1)), "bold.italic")),
        bg_params = list(fill=c(rep(c("white", "grey95"),
                                    length.out=(l-1)), "lightyellow")
        )))
  
df <- as.data.frame(npeak_combined_all[[run]])
df$hits <- as.factor(df$hits)
score_lim <- quantile(df$score, .99)
signalValue_lim <- quantile(df$signalValue, .98)
signalValue_min <- min(df$signalValue)
width_min <- quantile(df$max.width, .05)
width_max <- quantile(df$max.width, .95)
binsize <- ifelse(score_lim/40>10,score_lim/40,2)
  
g2 <- ggplot(df,aes(x=signalValue, y=score, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=1, position ='jitter', alpha=0.4) +
  scale_color_viridis_d(option ="viridis") +
  coord_cartesian(xlim=c(signalValue_min,signalValue_lim), ylim = c(0,score_lim)) +
  ggtitle(paste("C: score to signalValue"))

df2 <- df[df$hits == 1,]
df2$replicate <- str_split(df2$name, pattern ="_", simplify = T)[,2]
g3 <- ggplot(df2,aes(x=score, y=signalValue, color=replicate)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=1.5, position ='jitter', alpha=0.4) +
  scale_color_viridis_d(option ="turbo") +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(signalValue_min,signalValue_lim)) +
  ggtitle(paste("D: 1 hits"))

g3b <- ggplot(df2,aes(x=score, group=replicate, fill=replicate)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=1) + 
      # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
      scale_fill_viridis_d() +
      ggtitle(paste("D: scores per replicate (",n,")",sep="")) + coord_cartesian(xlim = c(0,score_lim))

g3c <- ggplot(df2,aes(x=replicate, y=score, fill=replicate)) +
  geom_violin() +
  geom_point(size=0.05, position ='jitter', alpha=0.1) +
  scale_fill_viridis_d(option ="viridis") +
  coord_cartesian(ylim = c(0,score_lim/4)) +
  ggtitle(paste("D: score per replicate"))

g5 <- ggplot(df,aes(x=max.width, y=score, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=1, position ='jitter', alpha=0.4) +
  scale_color_viridis_d(option ="viridis") +
  coord_cartesian(xlim=c(width_min,width_max), ylim = c(0,score_lim)) +
  ggtitle(paste("E: score to width"))

#  ht_list[[run]] <-  hist_hits[[run]] / gridExtra::tableGrob(table, theme = t1) +  plot_layout(heights = c(1,2))
  
g1 <- hist_hits[[run]]
g4 <- gridExtra::tableGrob(table, theme = t1)

paste("Method: ",run)
design <- ("ABB
           CDE")
plot(g1+g4+g2+g3c+g5+plot_layout(guides = "collect", axis_titles="collect", design=design))

ht_list[[run]] <-  (hist_hits[[run]] + g) / (g2 + gridExtra::tableGrob(table, theme = t1) + plot_layout(widths = c(1,1)))
}

# patchwork::wrap_plots(ht_list[1:2]) + plot_layout(ncol = 2)
# patchwork::wrap_plots(ht_list[3:4]) + plot_layout(ncol = 2)
# patchwork::wrap_plots(ht_list[1:5]) + plot_layout(ncol = 1)

# for (i in 1:5) {plot(ht_list[[i]])}

# plot peaks
i <- "pe"

#  bg_params = list(fill=c(rep("white",(l-1)), "skyblue1"))) 

```




#### Venn Overlap Peaks
```{r venn_overlaps, echo=FALSE}

# Add Gene names
ah = AnnotationHub()
query(ah, c("EnsDb", "Musculus"))
edb <- ah[["AH113713"]]

venn_top_list <- list()
venn_peak_list <- list()

# Venns
npeak_combined_1h <- list(npeak_combined_all[[1]][elementMetadata(npeak_combined_all[[1]])$hits > 1,],
                          npeak_combined_all[[2]][elementMetadata(npeak_combined_all[[2]])$hits > 1,],
                          npeak_combined_all[[3]][elementMetadata(npeak_combined_all[[3]])$hits > 1,],
                          npeak_combined_all[[4]][elementMetadata(npeak_combined_all[[4]])$hits > 1,],
                          npeak_combined_all[[5]][elementMetadata(npeak_combined_all[[5]])$hits > 1,])
names(npeak_combined_1h) <- names(npeak_combined_all)

length(npeak_combined_1h)
# pdf(paste(csrundir,"Venn_Peaks2.pdf",sep="/"))
v1 <- makeVennDiagram(main="Overlap of combined & filtered (>1 hits) peak lists",
                plot = TRUE, npeak_combined_1h,
                NameOfPeaks = names(npeak_combined_1h),
                fill=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
                cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise")
                )
v1ol <- findOverlapsOfPeaks(npeak_combined_1h)
v1_peaks <- v1ol[["peaklist"]][["Std///pe///local///chro5k///Online"]]
venn_top <- annotatePeak(v1_peaks, TxDb=txdb, tssRegion=c(-3000, 3000), verbose=FALSE) %>% as.data.frame()
  venn_top$transcriptId2 <- sub("\\.\\d+$", "", venn_top$transcriptId)
  venn_top$geneId <- mapIds(edb, keys = venn_top$transcriptId2, column = "GENEID", keytype = "TXID")
  venn_top$symbol <- mapIds(edb, keys = venn_top$transcriptId2, column = "SYMBOL", keytype = "TXID")
# colnames(venn_top)
 #  knitr::kable(venn_top[,c("symbol","distanceToTSS","annotation","seqnames","start","end")])  %>%
 # kable_styling("striped", full_width = F) %>% 
 # scroll_box(height = "400px")
venn_top_list[["v1"]] <- venn_top
venn_peak_list[["v1"]] <- v1_peaks
  
# Venns (with 1 hits)
# olvenn <- findOverlapsOfPeaks(npeak_combined_all)

length(npeak_combined_all)
# pdf(paste(csrundir,"Venn_Peaks2.pdf",sep="/"))
makeVennDiagram(main="Overlap of combined & filtered (with 1 hits) peak lists",
                plot = TRUE, npeak_combined_all,
                NameOfPeaks = names(npeak_combined_all),
                fill=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
                cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise")
                )
v2ol <- findOverlapsOfPeaks(npeak_combined_all)
v2_peaks <- v2ol[["peaklist"]][["Std///pe///local///chro5k///Online"]]
venn_top <- annotatePeak(v2_peaks, TxDb=txdb, tssRegion=c(-3000, 3000), verbose=FALSE) %>% as.data.frame()
  venn_top$transcriptId2 <- sub("\\.\\d+$", "", venn_top$transcriptId)
  venn_top$geneId <- mapIds(edb, keys = venn_top$transcriptId2, column = "GENEID", keytype = "TXID")
  venn_top$symbol <- mapIds(edb, keys = venn_top$transcriptId2, column = "SYMBOL", keytype = "TXID")
# colnames(venn_top)
 #  knitr::kable(venn_top[,c("symbol","distanceToTSS","annotation","seqnames","start","end")])  %>%
 # kable_styling("striped", full_width = F) %>% 
 # scroll_box(height = "400px")
venn_top_list[["v2"]] <- venn_top
venn_peak_list[["v2"]] <- v2_peaks

# Venns (3 Bowtie2 mapper)

length(npeak_combined_all)
# pdf(paste(csrundir,"Venn_Peaks2.pdf",sep="/"))
makeVennDiagram(main="Overlap of 3 Bowtie2 mapper",
                plot = TRUE, npeak_combined_all[1:3],
                NameOfPeaks = names(npeak_combined_all[1:3]),
                fill=c("#009E73", "lightgreen", "lightgreen"),
                cat.col=c("#009E73", "lightgreen", "lightgreen")
                )
v3ol <- findOverlapsOfPeaks(npeak_combined_all[1:3])
v3_peaks <- v3ol[["peaklist"]][["Std///pe///local"]]
venn_top <- annotatePeak(v3_peaks, TxDb=txdb, tssRegion=c(-3000, 3000), verbose=FALSE) %>% as.data.frame()
  venn_top$transcriptId2 <- sub("\\.\\d+$", "", venn_top$transcriptId)
  venn_top$geneId <- mapIds(edb, keys = venn_top$transcriptId2, column = "GENEID", keytype = "TXID")
  venn_top$symbol <- mapIds(edb, keys = venn_top$transcriptId2, column = "SYMBOL", keytype = "TXID")
# colnames(venn_top)
venn_top <- venn_top[order(abs(venn_top$distanceToTSS)),]
# knitr::kable(venn_top[1:20,c("symbol","distanceToTSS","annotation","seqnames","start","end")])  %>%
#  kable_styling("striped", full_width = F) %>% 
#  scroll_box(height = "400px")
venn_top_list[["v3"]] <- venn_top
venn_peak_list[["v3"]] <- v3_peaks
# dev.off()
# ggsave(peaks.venn,file=paste(pubfigdir,"Venn_Peaks.pdf",sep="/"), width = 8, height = 8, device = "pdf")


```
##### Venn top overlaps
```{r venn top, results="show"}

knitr::kable(venn_top_list[["v1"]][,c("symbol","distanceToTSS","annotation","seqnames","start","end")])  %>%
  kable_styling("striped", full_width = F) %>% 
  scroll_box(height = "400px")

knitr::kable(venn_top_list[["v2"]][,c("symbol","distanceToTSS","annotation","seqnames","start","end")])  %>%
  kable_styling("striped", full_width = F) %>% 
  scroll_box(height = "400px")

knitr::kable(venn_top_list[["v3"]][1:20,c("symbol","distanceToTSS","annotation","seqnames","start","end")])  %>%
  kable_styling("striped", full_width = F) %>% 
  scroll_box(height = "400px")


```



## Annotate Peaks
#### 1 Run
```{r annotated_peaks, fig.width=12, fig.height=10, include=TRUE, results="show", out.width="100%"}

# Add Gene names
ah = AnnotationHub()
query(ah, c("EnsDb", "Musculus"))
edb <- ah[["AH113713"]]

# 1 Subfolder

figures_annotated_peaks <- list()

i <- "pe"
all_npeaksAnno <- annotatePeak(npeak_combined_all[[i]], TxDb=txdb,tssRegion=c(-3000, 3000), verbose=FALSE)
  all_npeaksAnno_table <- as.data.frame(all_npeaksAnno)
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]

# Annotated table
top_hits <- all_npeaksAnno_table[order(all_npeaksAnno_table$signalValue, decreasing=T),]
colnames(top_hits)
kable(top_hits[1:15,c("symbol","distanceToTSS","annotation_short","score","signalValue","hits","seqnames","start","end","width","annotation","geneId")])  %>%
 kable_styling("striped", full_width = F) %>% 
 scroll_box(height = "400px")

hits_lim <- max(all_npeaksAnno_table$hits)
score_lim <- quantile(all_npeaksAnno_table$score, .99)
signalValue_lim <- quantile(all_npeaksAnno_table$signalValue, .99)
qValue_lim <- quantile(all_npeaksAnno_table$qValue, .90)

g1 <- ggplot(all_npeaksAnno_table,aes(x=distanceToTSS, y=score, color=hits)) +
  geom_point(size=1, alpha=0.5, position = "jitter") +
  scale_color_viridis_c() +
  coord_cartesian(ylim=c(0,score_lim)) +
  ggtitle(paste("scores to distance"))

g2 <- ggplot(all_npeaksAnno_table,aes(x=distanceToTSS, y=signalValue, color=hits)) +
  geom_point(size=1, alpha=0.5, position = "jitter") +
  scale_color_viridis_c() +
  coord_cartesian(ylim=c(1,signalValue_lim)) +
  ggtitle(paste("signalValue to distance"))

g3 <- ggplot(all_npeaksAnno_table,aes(x=distanceToTSS)) +
      stat_bin(alpha=0.6, position = 'identity', binwidth=1000, fill=viridis(10)[3]) +
      coord_cartesian(xlim = c(-50000,+50000))
g4 <- ggplot(all_npeaksAnno_table,aes(x=distanceToTSS)) +
      stat_bin(alpha=0.6, position = 'identity', binwidth=100, fill=viridis(10)[5]) +
      coord_cartesian(xlim = c(-5000,+5000)) 

# ( (g1+g2) / 
#   (g1 + coord_cartesian(xlim=c(-500000,+500000),ylim=c(0,score_lim)) + 
#     g2 + coord_cartesian(xlim=c(-500000,+500000),ylim=c(1,signalValue_lim))) /
#   (g1 + coord_cartesian(xlim=c(-50000,+50000),ylim=c(0,score_lim)) + 
#     g2 + coord_cartesian(xlim=c(-50000,+50000),ylim=c(1,signalValue_lim))) /
#   (g3 + g4)
#   ) + plot_layout(guides = "collect", axis_titles="collect")

 (g1 + 
     (g1 + coord_cartesian(xlim=c(-500000,+500000),ylim=c(0,score_lim)))) /
  ((g1 + coord_cartesian(xlim=c(-50000,+50000),ylim=c(0,score_lim))) + 
         (g1 + coord_cartesian(xlim=c(-5000,+5000),ylim=c(0,score_lim)))) /
  (g3 + g4) + plot_layout(guides = "collect", axis_titles="collect")

```



```{r annotae_peak1_old, eval=FALSE, include=FALSE}

top_hits <- top_hits[order(top_hits$score, decreasing=T),]


top_hits <- subset(all_npeaksAnno_table, distanceToTSS > -2000 & distanceToTSS < 2000) 
dim(top_hits)
top_hits <- top_hits[order(top_hits$score, decreasing=T),]
knitr::kable(top_hits[c(0:50),c("symbol","annotation","distanceToTSS","peak","score","signalValue","qValue","hits")],format = "markdown") %>%
 kable_styling("striped", full_width = F) %>% 
 scroll_box(height = "400px")

# extraCols_narrowPeak <- c(signalValue = "numeric", pValue = "numeric",
                          # qValue = "numeric", peak = "integer")

# 5th: "score" = int(-10*log10qvalue)
# 6th: "strand" = [empty]
# 7th: fold-change at peak summit
# 8th: "pValue" = -log10pvalue at peak summit
# 9th: "qValue" = -log10qvalue at peak summit
# 10th: "peak" = - relative summit position to peak start


hits_lim <- max(all_npeaksAnno_table$hits)
score_lim <- quantile(all_npeaksAnno_table$score, .90)
peak_lim <- quantile(all_npeaksAnno_table$peak, .90)
signalValue_lim <- quantile(all_npeaksAnno_table$signalValue, .90)
qValue_lim <- quantile(all_npeaksAnno_table$qValue, .90)


g1 <- ggplot(all_npeaksAnno_table,aes(x=score, y=peak, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(0,peak_lim)) +
  ggtitle(paste("score to peak"))

g2 <- ggplot(all_npeaksAnno_table,aes(x=score, y=signalValue, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(1.9,signalValue_lim)) +
  ggtitle(paste("score to signalValue"))

g3 <- ggplot(all_npeaksAnno_table,aes(x=score, y=qValue, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(1.9,qValue_lim)) +
  ggtitle(paste("score to qValue"))


g4 <- ggplot(all_npeaksAnno_table,aes(x=peak, y=signalValue, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,peak_lim), ylim = c(1.9,signalValue_lim)) +
  ggtitle(paste("peak to signalValue"))

g1+g2+g3+g4+plot_layout(guides = "collect")


kable(head(all_npeaksAnno_table))  %>%
 kable_styling("striped", full_width = F) %>% 
 scroll_box(height = "400px")

ggplot(all_npeaksAnno_table,aes(x=distanceToTSS, y=score, color=hits)) +
  geom_point(size=1, alpha=0.5, position = "jitter") +
  scale_color_viridis_c() +
  coord_cartesian(ylim=c(0,score_lim)) +
  ggtitle(paste("scores to signalValue"))
# xlim=c(-10000,10000)


```

#### All runs
```{r annotae_all_peaks, results="show", fig.width=12, out.width="100%", fig.height=9}
i <- 2
top_hits_list <- list()
for (i in 1:length(npeak_combined_all)){
all_npeaksAnno <- annotatePeak(npeak_combined_all[[i]], TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
  all_npeaksAnno_table <- as.data.frame(all_npeaksAnno)
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
  all_npeaksAnno_table$hits <- as.factor(all_npeaksAnno_table$hits)

n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]

hits_lim <- max(levels(all_npeaksAnno_table$hits))
score_lim <- quantile(all_npeaksAnno_table$score, .99)
peak_lim <- quantile(all_npeaksAnno_table$peak, .99)
signalValue_lim <- quantile(all_npeaksAnno_table$signalValue, .99)
qValue_lim <- quantile(all_npeaksAnno_table$qValue, .99)
distanceToTSS_max <- quantile(all_npeaksAnno_table$distanceToTSS, .90)
distanceToTSS_min <- quantile(all_npeaksAnno_table$distanceToTSS, .1)
width_lim <- quantile(all_npeaksAnno_table$width, .99)
signalValue_min <- min(all_npeaksAnno_table$signalValue)

df <- all_npeaksAnno_table

g1a <- ggplot(df,aes(x=distanceToTSS, y=signalValue, color=hits)) +
  geom_point(size=1.5, position ='jitter', alpha=0.4) +
  scale_color_viridis_d(option ="viridis") +
  coord_cartesian(xlim=c(distanceToTSS_min,distanceToTSS_max), ylim = c(signalValue_min,signalValue_lim)) +
  ggtitle(paste("A: signalValue to distance"))

g1b <- ggplot(df,aes(x=distanceToTSS, group=hits, fill=hits)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=c(10000/100)) + 
      # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
      scale_fill_viridis_d() +
      ggtitle(paste("B: hits per distance (",n,")",sep="")) + coord_cartesian(xlim = c(-5000, 5000))

g2a <- ggplot(df,aes(x=width, y=signalValue, color=hits)) +
  geom_point(size=1.5, position ='jitter', alpha=0.4) +
  scale_color_viridis_d(option ="viridis") +
  coord_cartesian(xlim=c(180,width_lim), ylim = c(signalValue_min,signalValue_lim)) +
  ggtitle(paste("C: signalValue per peak width (hits)"))

g2b <- ggplot(df,aes(x=width, y=signalValue, color=annotation_short)) +
  geom_point(size=1.5, position ='jitter', alpha=0.4) +
  scale_color_viridis_d(option ="turbo") +
  coord_cartesian(xlim=c(180,width_lim), ylim = c(signalValue_min,signalValue_lim)) +
  ggtitle(paste("D: signalValue per peak width (region)"))

g2c <- ggplot(df,aes(x=width, y=signalValue, color=distanceToTSS)) +
  geom_point(size=1.5, position ='jitter', alpha=0.4) +
  scale_color_viridis_c(option ="turbo", limits = c(distanceToTSS_min*0.2, distanceToTSS_max*0.2)) +
  coord_cartesian(xlim=c(0,width_lim), ylim = c(signalValue_min,signalValue_lim)) +
  ggtitle(paste("E: signalValue per peak width (region)"))


figures_annotated_peaks[[n]] <- g1a+g1b+g2a+g2b+
                                  plot_layout(ncol = 2) + 
                                  plot_annotation(title = paste("Method:",n))

# axis_titles = "collect",guides = "collect"

top_hits <- subset(all_npeaksAnno_table, distanceToTSS > -2000 & distanceToTSS < 2000) 
dim(top_hits)
top_hits <- top_hits[order(top_hits$score, decreasing=T),]
top_hits_list[[n]] <- top_hits

header = 7 
names(header) = paste("Method: ",n)

print(cat("\n"))
 
plot(figures_annotated_peaks[[n]])

}

# top_hits_list
# figures_annotated_peaks


knitr::kable(top_hits_list[[1]][c(0:10),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")  %>% kable_styling("striped", full_width = F) %>% 
 add_header_above( header = header) %>%
  scroll_box(height = "400px")

knitr::kable(top_hits_list[[2]][c(0:10),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")  %>% kable_styling("striped", full_width = F) %>% 
 add_header_above( header = header) %>%
  scroll_box(height = "400px")

knitr::kable(top_hits_list[[3]][c(0:10),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")  %>% kable_styling("striped", full_width = F) %>% 
 add_header_above( header = header) %>%
  scroll_box(height = "400px")

knitr::kable(top_hits_list[[4]][c(0:10),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")  %>% kable_styling("striped", full_width = F) %>% 
 add_header_above( header = header) %>%
  scroll_box(height = "400px")

knitr::kable(top_hits_list[[5]][c(0:10),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")  %>% kable_styling("striped", full_width = F) %>% 
 add_header_above( header = header) %>%
  scroll_box(height = "400px")


```




#### All runs old2
```{r annotae_all_peaks_old2, eval=FALSE, fig.width=12, include=FALSE, out.width="100%"}


top_hits_list <- list()
for (i in 1:length(npeak_combined_all)){
all_npeaksAnno <- annotatePeak(npeak_combined_all[[i]], TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
  all_npeaksAnno_table <- as.data.frame(all_npeaksAnno)
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]



hits_lim <- max(all_npeaksAnno_table$hits)
score_lim <- quantile(all_npeaksAnno_table$score, .95)
peak_lim <- quantile(all_npeaksAnno_table$peak, .95)
signalValue_lim <- quantile(all_npeaksAnno_table$signalValue, .95)
qValue_lim <- quantile(all_npeaksAnno_table$qValue, .95)

g1 <- ggplot(all_npeaksAnno_table,aes(x=hits)) + # fill = cut(hits, 100)
      stat_bin(alpha=0.6, position = 'identity', binwidth=1, fill=viridis(10)[2]) + 
      stat_bin(position = 'identity', binwidth=1, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
      ggtitle("A: peaks in replicates")

g2a <- ggplot(all_npeaksAnno_table,aes(x=score, group=annotation_short, fill=annotation_short)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=5) + 
      # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
      scale_fill_viridis_d() +
      ggtitle(paste("genetic region per score (",n,")",sep="")) + coord_cartesian(xlim = c(0, score_lim))

g2b <- ggplot(all_npeaksAnno_table,aes(x=signalValue, group=annotation_short, fill=annotation_short)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=1) + 
  # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
  scale_fill_viridis_d() +
  ggtitle(paste("genetic region per signalValue (",n,")",sep="")) + coord_cartesian(xlim = c(0, 15))

g3 <- ggplot(all_npeaksAnno_table,aes(x=width, group=hits, fill=hits)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=200) + 
  # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
  scale_fill_viridis_c() +
  ggtitle(paste("hits per peak size")) + coord_cartesian(xlim = c(0, 10000))

all_npeaksAnno_table$hits <- as.factor(all_npeaksAnno_table$hits)

# g4 <- ggplot(all_npeaksAnno_table,aes(x=hits, y=qValue)) +
#   geom_violin(aes(fill=hits)) +
#   scale_fill_viridis_d() +
#   geom_point(size=0.05, position ='jitter', alpha=0.2,color="grey40") +
#   coord_cartesian(ylim=c(0,50))

g4 <- ggplot(all_npeaksAnno_table,aes(x=hits, y=score)) +
  geom_violin(aes(fill=hits, color=hits)) +
  scale_fill_viridis_d() +
  geom_point(size=0.05, position ='jitter', alpha=0.2,color="grey40") +
  coord_cartesian(ylim=c(0,1000)) +
  ggtitle(paste("scores per hit"))

levels(factor(all_npeaksAnno_table$annotation_short))

g5 <- ggplot(all_npeaksAnno_table,aes(x=score, y=qValue, color=signalValue)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c(limits = c(0, 15)) +
  coord_cartesian(ylim=c(0,50), xlim = c(0,500)) +
  ggtitle(paste("score to qvalue"))

g6 <- ggplot(all_npeaksAnno_table,aes(x=score, y=signalValue, color=pValue)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=1, alpha=0.5) +
  scale_color_viridis_c(option = "D", direction = -1, limits = c(0, 100)) +
  coord_cartesian(ylim=c(0,15), xlim = c(0,1000))+
  ggtitle(paste("scores to signalValue"))

# datatable(head(all_npeaksAnno_table[order(all_npeaksAnno_table$width),]))
viridis(8)

top_hits <- subset(all_npeaksAnno_table, distanceToTSS > -2000 & distanceToTSS < 2000) 
dim(top_hits)
top_hits <- top_hits[order(top_hits$score, decreasing=T),]
top_hits_list[[n]] <- top_hits

figures_annotated_peaks[[n]] <- g1+g2a+g3+g4+g5+g6+plot_layout(nrow = 2, ncol = 3, axis_titles = "collect") + plot_annotation(title = paste("Method:",n))
}

figures_annotated_peaks

options(kableExtra.auto_format = FALSE)
knitr::kable(top_hits[c(0:50),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")

# datatable(top_hits[c(0:50),c("symbol","distanceToTSS","hits","score","signalValue","qValue")])

```

#### All runs (generate lists)
```{r all_peaks_old3, eval=TRUE, include=FALSE}

## All peaks
  
npeakAnnoList <- lapply(npeak_combined_all, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)

# make list
npeakAnnoList_table <- list()
gene_ids_all_list <- list()
for (i in 1:length(npeakAnnoList)){
  print(paste(i,": ",names(npeakAnnoList)[i],sep=""))
  n <- names(npeakAnnoList[i])
  table <- as.data.frame(npeakAnnoList[[i]]@anno)
  table$transcriptId2  <- sub("\\.\\d+$", "", table$transcriptId)
  table$entrez <- table$geneId
  table$geneId <- mapIds(edb, keys = table$transcriptId2, column = "GENEID", keytype = "TXID")
  table$symbol <- mapIds(edb, keys = table$transcriptId2, column = "SYMBOL", keytype = "TXID")
  assign(paste("narrow_table",n,sep="_"),table)
  table_filter <- subset(table, distanceToTSS > -2000 & distanceToTSS < 2000)
  gene_ids_all_list <- lapply(gene_ids_all_list,unique)

  print(paste("Before: ",length(rownames(table)),", After: ",length(rownames(table_filter))," (",round(length(rownames(table_filter))/length(rownames(table))*100),"%)"))
  npeakAnnoList_table[[n]] <- table_filter
  gene_ids_all_list[[n]] <- table_filter$geneId
}

# Results from RNA-Seq
deg <- readRDS("../../RNA-Seq_res_degs.RDS")
deg.top <- readRDS("../../RNA-Seq_top_degs.RDS")

# Plot gene region for all lists
i <- 5
for (i in names(npeakAnnoList)){

  all_npeaksAnno_table <- as.data.frame(npeakAnnoList[[i]])
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]
}


```

#### Venns
```{r eval=FALSE, include=FALSE}
# Results from RNA-Seq
deg <- readRDS("../../RNA-Seq_res_degs.RDS")
deg.top <- readRDS("../../RNA-Seq_top_degs.RDS")

# overlap
gene_ids_all_list[["RNASeq"]] <- deg.top$gene_id

id_comb <- unlist(gene_ids_all_list) %>% unique()
id_symb <- mapIds(edb, keys = id_comb, column = "SYMBOL", keytype = "GENEID", multiVals = "first")

id_com_table <- data.frame("ensembl" = id_comb,
                           "id_symbol" = id_symb) 

for (n in names(gene_ids_all_list)){
  isin <- {}
  for (i in id_com_table$ensembl){
    ifelse(i %in% gene_ids_all_list[[n]],
         isin <- c(isin,1),
         isin <- c(isin,0))
  }
  id_com_table <- cbind(id_com_table,isin)
  l <- ncol(id_com_table)
  colnames(id_com_table)[l] <- n
}
id_com_table

subset(id_com_table, chro5k == 1)
subset(id_com_table, Std == 1)
subset(id_com_table, pe == 1 & local == 1 & Online == 0 & RNASeq == 1)$id_symbol

# & Online == TRUE
# & RNASeq == TRUE

id_com_table$sum <- rowSums(id_com_table[,3:8])
summary(duplicated(id_com_table$id_symbol))
id_com_table <- id_com_table[!duplicated(id_com_table$id_symbol),]
id_com_table <- id_com_table[!is.na(id_com_table$id_symbol),]
id_com_table <- id_com_table[order(id_com_table$sum, decreasing = TRUE),]
rownames(id_com_table) <- id_com_table$id_symbol

heatmap(Rowv = TRUE, as.matrix(subset(id_com_table,sum>1)[,3:8]))

pdf(file=paste(csdir,"/Heatmap.pdf",sep=""))
heatmap(as.matrix(subset(id_com_table,Std == 1 | chro5k == 1 | pe == 1 & local ==1)[,3:8]),
        Rowv = TRUE)
heatmap(as.matrix(subset(id_com_table,Std == 1 | chro5k == 1 | pe == 1 | local ==1)[,3:8]),
        Rowv = FALSE)
dev.off()
# ggsave(file=paste(pubfigdir,"/Heatmap.png",sep=""), width = 8, height = 8)

heat.data <- subset(id_com_table,Std == 1 | chro5k == 1 | pe == 1 & local ==1)[,3:8]
ggplot(heat.data, aes(x=c(colnames(heat.data)),y=rownames(), fill= Z)) + 
  geom_tile()

lapply(gene_ids_all_list,length)
gene_ids_all_list <- lapply(gene_ids_all_list,unique)
lapply(gene_ids_all_list,length)

length(gene_ids_all_list)
vennplot(gene_ids_all_list[1:5])
ggvenn(gene_ids_all_list[1:4],
      fill_color = c("yellow3", "yellow", "yellow","darkolivegreen1"),
      stroke_size = 0.5,
      show_percentage = FALSE)
ggsave(file=paste(pubfigdir,"/Venn_Genes_own.png",sep=""), width = 8, height = 8)

in_pe_local <- intersect(gene_ids_all_list[["pe"]],
          gene_ids_all_list[["local"]])
in_pe_local_Std <- intersect(in_pe_local,gene_ids_all_list[["Std"]])
in_pe_local_Std

,
          gene_ids_all_list[["pe"]]
,
          gene_ids_all_list[["local"]],
          gene_ids_all_list[["Online"]],
          gene_ids_all_list[["chro5k"]],

)


setdiff (intersect(intersect(gene_ids_all_list["chro5k"], gene_ids_all_list["Std"]), UPDEGsTR40SR40genes),  UPDEGsTRCSRCgenes)


ggvenn(gene_ids_all_list[c(2,3,5,6)],
      fill_color = c("yellow", "yellow","lightblue","salmon"),
      stroke_size = 0.5,
      show_percentage = FALSE)
ggsave(file=paste(pubfigdir,"/Venn_Genes_compare.png",sep=""), width = 8, height = 8)

ol <- calculate.overlap(gene_ids_all_list[1:5])


       

library(ggplot2)
library(ggVennDiagram)

x <- list(A=1:5,B=2:7,C=3:6,D=4:9)




# manual
venn <- Venn(x)
data <- process_data(venn)
ggplot() +
  # 1. region count layer
  geom_sf(aes(fill = count), data = venn_region(data)) +
  # 2. set edge layer
  geom_sf(aes(color = name), data = venn_setedge(data), show.legend = TRUE, size = 2) +
  # 3. set label layer
  geom_sf_text(aes(label = name), data = venn_setlabel(data)) +
  # 4. region label layer
  geom_sf_label(aes(label = paste0(count, " (", scales::percent(count/sum(count), accuracy = 2), ")")), 
                data = venn_region(data),
                size = 3) +
  scale_fill_gradient(low = "#F4FAFE", high = "#4981BF")+
  scale_color_manual(values = c("A" = "yellow","B" ="steelblue",'C' = 'red', 'D' = 'black'),
                     labels = c('D' = 'D = bdiv_human'))+
  theme_void()




v <- ggVennDiagram(gene_ids_all_list,
              label_alpha = 0,
              label = "count",
              set_color = c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"))
v + scale_fill_distiller(palette = "RdBu")

              ,
              
              scale_fill_distiller(palette = "RdBu"))
,
              scale_fill_manual = c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
              cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"))
              )

makeVennDiagram(gene_ids_all_list)

,by=gplots)

                         fill=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
                cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise")
                )
       
       
ol <- calculate.overlap(gene_ids_all_list[1:3])
summary(ol)
ol <- calculate.overlap(list("ChIP.all"=subset(gene_table_hits, hits > 7)$geneid,
                             "RNA-Seq"=deg.top$gene_id))



venn.diagram(gene_ids_all_list)

chip.rna <- sort(unname(mapIds(edb,keys=ol$a3,column = "SYMBOL",keytype = "GENEID", multiVals = "first")))

beep()


```



# Visualize Peaks

### prepare bams

```{r, eval=TRUE, include=FALSE}
# bams
bdir <- paste(ddir,"Bowtie2",sep="/")
sbdirs <- list.dirs(bdir)[-c(1,2)]
bams <- {}
bamfiles_list <- list()
for (d in sbdirs) {
  n <- basename(d)
  bamfiles <- list.files(d,pattern=".bam$")
  bamfiles <- paste(d,bamfiles,sep="/")
  bams <- c(bams,bamfiles)
  bamfiles_list[[n]] <- bamfiles
}
bamfiles_list
bamfiles
bams

# peaks
# npeak_combined_all

# Genes
# npeakAnnoList[[1]]@anno

# Gene Table




```

### prepare bigwigs

```{r}
# bigwigs
bigwigdir <- paste(ddir,"deeptools/bigwig",sep="/")
sbwdirs <- list.dirs(bigwigdir)[-1]
bigwigs <- {}
bwfiles_list <- list()
for (d in sbwdirs) {
  n <- basename(d)
  bwfiles <- list.files(d,pattern=".bw$")
  bwfiles <- paste(d,bwfiles,sep="/")
  bigwigs <- c(bigwigs,bwfiles)
  bwfiles_list[[n]] <- bwfiles
}
bwfiles_list
bigwigs

```


## prepare txdb data
```{r}

# browseIntervals(npeak_combined_all[["Std"]][1:5])

# 
txdb <- TxDb.Mmusculus.UCSC.mm39.knownGene
gene_models <- genes(txdb)
colnames(edb)
keytypes(edb)
gene_models$ensembl <- mapIds(edb, keys = gene_models$gene_id, column = "GENEID", keytype = "ENTREZID")
gene_models$symbol <- mapIds(edb, keys = gene_models$gene_id, column = "SYMBOL", keytype = "ENTREZID")
AOC1 <- gene_models["76507"]
# Plotting window: 500 bp around AKAP2
gene_window <- AOC1 + 500
# Window around promoter for second examples
promoter_window <- promoters(AOC1, upstream=500, downstream=500)
tx_models <- exonsBy(txdb, by="tx", use.names=TRUE)
meta_models <- exonsBy(txdb, by="gene")

```


## trackViewer
#### example
```{r example, eval=FALSE, include=TRUE}
## trackViewer example
extdata <- system.file("extdata", package="trackViewer",
                       mustWork=TRUE)
gr <- GRanges("chr11", IRanges(122929275, 122930122), strand="-")
fox2 <- importScore(file.path(extdata, "fox2.bed"), format="BED",
                    ranges=gr)
fox2$dat <- coverageGR(fox2$dat)

viewTracks(trackList(fox2), gr=gr, autoOptimizeStyle=TRUE, newpage=TRUE)

dt <- DataTrack(range=fox2$dat[strand(fox2$dat)=="-"] , 
                genome="hg19", type="hist", name="fox2", 
                window=-1, chromosome="chr11", 
                fill.histogram="black", col.histogram="NA",
                background.title="white",
                col.frame="white", col.axis="black",
                col="black", col.title="black")
plotTracks(dt, from=122929275, to=122930122, strand="-")


# Maunals
## tidyGenomeBrowser
## https://github.com/MalteThodberg/tidyGenomeBrowser/blob/master/README.Rmd

## trackViewer
## https://bioconductor.org/packages/release/bioc/vignettes/trackViewer/inst/doc/trackViewer.html

## Generate bigwig
## https://hbctraining.github.io/Intro-to-ChIPseq-flipped/lessons/08_creating_bigwig_files.html

## Normalize bigwig
## https://www.biostars.org/p/394434/

```
#### own data test 1 gene
```{r plot_tracks_1, results="show"}
# # Step 1. Import data
# ## example
# repA <- importScore(file.path(extdata, "cpsf160.repA_-.wig"),
#                     file.path(extdata, "cpsf160.repA_+.wig"),
#                     format="WIG")
# ## Because the wig file does not contain any strand info, 
# ## we need to set it manually.
# strand(repA$dat) <- "-"
# strand(repA$dat2) <- "+"
# 
# # Step 2. Build the gene model
# library(TxDb.Hsapiens.UCSC.hg19.knownGene)
# library(org.Hs.eg.db)
# gr <- GRanges("chr11", IRanges(122929275, 122930122), strand="-")
# trs <- geneModelFromTxdb(TxDb.Hsapiens.UCSC.hg19.knownGene,
#                          org.Hs.eg.db,
#                          gr=gr)
# # Step 3. View the tracks
# viewerStyle <- trackViewerStyle()
# setTrackViewerStyleParam(viewerStyle, "margin", c(.1, .05, .02, .02))
# trackList <- trackList(trs)
# trackList <- trackList(repA, trs)
# vp <- viewTracks(trackList, 
#                  gr=gr, viewerStyle=viewerStyle, 
#                  autoOptimizeStyle=TRUE)

# own data
# get Peak info

top_hits_genes <- subset(top_hits_list[["pe"]], distanceToTSS > -2000 & distanceToTSS < 2000, hits > 1) 

peak <- top_hits_genes[2,]
gr <- GRanges(peak[,1:3])+1000
gr$peak <- peak$start+peak$peak
trs <- geneModelFromTxdb(txdb,
                         org.Mm.eg.db,
                         gr=gr)

bw_pe_1580 <- importScore(bigwigs[1],
                          ranges = gr,
                    format="BigWig")

bw_pe_1580_1578 <- importScore(bigwigs[3],
                          ranges = gr,
                    format="BigWig")

bw_pe_1580_2 <- importScore(bigwigs[2],
                          ranges = gr,
                    format="BigWig")

bw_pe_1580_1578_4 <- importScore(bigwigs[4],
                          ranges = gr,
                    format="BigWig")


bam_track_list <- list()
for (i in bams[25:42]){
  bam_track <- importBam(i,
                          ranges = gr)
  bam_track_list[[i]] <- bam_track
}
# bam_track_list

# Step 3. View the tracks
viewerStyle <- trackViewerStyle()
setTrackViewerStyleParam(viewerStyle, "margin", c(.1, .05, .02, .02))
trackList <- trackList(trs)
trackList <- trackList(bam_track_list,trs)
vp <- viewTracks(trackList, 
                 gr=gr, viewerStyle=viewerStyle, 
                 autoOptimizeStyle=TRUE)
addGuideLine(gr$peak, vp=vp, col="red")


```

### plot all peaks of top hits
```{r trackviewer_loop, eval=TRUE}
top_hits_genes
i <- 8
for (i in 1:10){

peak <- top_hits_genes[i,]
print(peak$symbol)
gr <- GRanges(peak[,1:3])+2000
gr$peak <- peak$start+peak$peak
trs <- {}
trs <- geneModelFromTxdb(txdb,
                         org.Mm.eg.db,
                         gr=gr)

# bam_track_list <- list()
# max <- {}
# # for (i in bams[25:42]){
# #   bam_track <- importBam(i,
# #                           ranges = gr)
# #   bam_track_list[[i]] <- bam_track
# #   max <- max(c(max,bam_track@dat@elementMetadata@listData[["score"]]))
# # }

# parallel loop
n.cores <- 10
my.cluster <- parallel::makeCluster(n.cores,type = "FORK")
doParallel::registerDoParallel(cl = my.cluster)
foreach::getDoParRegistered()
foreach::getDoParWorkers()

comb <- function(x, ...) {
  lapply(seq_along(x),
    function(i) c(x[[i]], lapply(list(...), function(y) y[[i]])))
}

max <- {}
oper <- foreach(i=25:42, .combine='comb', .multicombine=TRUE,
                .init=list(list(), list())) %dopar% {
  b <- bams[i]
  bam_track <- importBam(b,ranges = gr)
  max <- max(bam_track@dat@elementMetadata@listData[["score"]])
  list(bam_track,max)
  }
parallel::stopCluster(cl = my.cluster)
bam_track_list <- oper[[1]]
max <- max(unlist(oper[[2]]))

names(bam_track_list) <- str_split(bams, pattern = "_", simplify = T)[25:42,7]


# Step 3. View the tracks
viewerStyle <- trackViewerStyle()
setTrackViewerStyleParam(viewerStyle, "margin", c(.1, .05, .02, .02))

trackList <- trackList(trs)
trackList <- trackList(bam_track_list,trs)

optSty <- optimizeStyle(trackList(bam_track_list,trs))
trackList <- optSty$tracks
viewerStyle <- optSty$style
setTrackXscaleParam(trackList[[1]], "draw", TRUE)
setTrackXscaleParam(trackList[[1]], "gp", list(cex=0.8))
setTrackViewerStyleParam(viewerStyle, "margin", c(.01, .05, .01, .05))
for(i in 1:length(trackList)){
    setTrackYaxisParam(trackList[[i]], "main", FALSE)}
## Adjust the limit of y-axis
for(i in 1:length(trackList)){
    setTrackStyleParam(trackList[[i]], "ylim", c(0, max))}
vp <- viewTracks(trackList, 
                 gr=gr, viewerStyle=viewerStyle, 
                 autoOptimizeStyle=TRUE)
addGuideLine(gr$peak, vp=vp, col="red")
}

# browseTracks(trackList, gr=gr)


```

### plot all peaks of candidates
```{r trackviewer_loop, eval=TRUE}

venn_top_list[["v3"]]
venn_peak_list[["v2"]]
peaks <- venn_peak_list[["v2"]]
l <- length(peaks)
i <- 8
for (i in 1:l){

peak <- peaks[i,]
# print(peak$symbol)
gr <- GRanges(peak[,1:3])+2000
gr$peak <- peak$start+peak$peak
trs <- {}
trs <- geneModelFromTxdb(txdb,
                         org.Mm.eg.db,
                         gr=gr)

# bam_track_list <- list()
# max <- {}
# # for (i in bams[25:42]){
# #   bam_track <- importBam(i,
# #                           ranges = gr)
# #   bam_track_list[[i]] <- bam_track
# #   max <- max(c(max,bam_track@dat@elementMetadata@listData[["score"]]))
# # }

# parallel loop
n.cores <- 10
my.cluster <- parallel::makeCluster(n.cores,type = "FORK")
doParallel::registerDoParallel(cl = my.cluster)
foreach::getDoParRegistered()
foreach::getDoParWorkers()

comb <- function(x, ...) {
  lapply(seq_along(x),
    function(i) c(x[[i]], lapply(list(...), function(y) y[[i]])))
}

max <- {}
oper <- foreach(i=25:42, .combine='comb', .multicombine=TRUE,
                .init=list(list(), list())) %dopar% {
  b <- bams[i]
  bam_track <- importBam(b,ranges = gr)
  max <- max(bam_track@dat@elementMetadata@listData[["score"]])
  list(bam_track,max)
  }
parallel::stopCluster(cl = my.cluster)
bam_track_list <- oper[[1]]
max <- max(unlist(oper[[2]]))

names(bam_track_list) <- str_split(bams, pattern = "_", simplify = T)[25:42,7]


# Step 3. View the tracks
viewerStyle <- trackViewerStyle()
setTrackViewerStyleParam(viewerStyle, "margin", c(.1, .05, .02, .02))

trackList <- trackList(trs)
trackList <- trackList(bam_track_list,trs)

optSty <- optimizeStyle(trackList(bam_track_list,trs))
trackList <- optSty$tracks
viewerStyle <- optSty$style
setTrackXscaleParam(trackList[[1]], "draw", TRUE)
setTrackXscaleParam(trackList[[1]], "gp", list(cex=0.8))
setTrackViewerStyleParam(viewerStyle, "margin", c(.01, .05, .01, .05))
for(i in 1:length(trackList)){
    setTrackYaxisParam(trackList[[i]], "main", FALSE)}
## Adjust the limit of y-axis
for(i in 1:length(trackList)){
    setTrackStyleParam(trackList[[i]], "ylim", c(0, max))}
vp <- viewTracks(trackList, 
                 gr=gr, viewerStyle=viewerStyle, 
                 autoOptimizeStyle=TRUE)
addGuideLine(gr$peak, vp=vp, col="red")
}

# browseTracks(trackList, gr=gr)


```


```{bash eval=FALSE, include=FALSE}

cd /mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/deeptools$ 


```



