---
title: "ChIP_Wt1 - Results"
author: "Kelterborn"
date: "`r Sys.Date()`"
output:
  md_document:
    variant: gfm
always_allow_html: true
---

<!-- <style> -->
<!-- .vscroll-plot { -->
<!--     width: 1000px; -->
<!--     height: 1000px; -->
<!--     overflow-y: scroll; -->
<!--     overflow-x: scroll; -->
<!-- } -->
<!-- </style> -->

<!-- output: -->
<!--   html_document:  -->
<!--     toc: true -->
<!--     keep_md: true -->
<!--     self_contained: false -->
<!--     df_print: kable -->


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      results = "hide", error=FALSE, warning=FALSE, message=FALSE)
options(kableExtra.auto_format = FALSE)
```

# R Prepare System
## R update and load librarys
BiocManager::install("")

```{r librarys}
# BiocManager::install(update = TRUE, ask = FALSE)

library(dbplyr)
library(tidyverse)
library(ChIPseeker)
library(rtracklayer)
library(trackViewer)
library(GenomicRanges)
library(IRanges)
library(ChIPpeakAnno)
library(AnnotationHub)
library(ggplot2)
library(viridis)
library(kableExtra)
library(DT)
library(patchwork)
library(gridExtra)

library(TxDb.Mmusculus.UCSC.mm39.knownGene)
txdb <- TxDb.Mmusculus.UCSC.mm39.knownGene

```

```{r old_librarys, eval=FALSE, include=FALSE}
# library("ggVennDiagram")
# library("ggvenn")
# 

# library(ChIPQC)
# library(BiocParallel)
# library(readxl)
# library(writexl)
# library(xlsx)
# library(beepr)

# library(TxDb.Hsapiens.UCSC.hg19.knownGene)
# library(TxDb.Hsapiens.UCSC.hg38.knownGene)
# library(TxDb.Mmusculus.UCSC.mm10.knownGene)
# library(TxDb.Mmusculus.UCSC.mm39.knownGene)
# # txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
# # txdb <- TxDb.Mmusculus.UCSC.mm10.knownGene
# txdb <- TxDb.Mmusculus.UCSC.mm39.knownGene
# library(clusterProfiler)
# library(UpSetR)
# library(ggimage)
# library(ReactomePA)
# library(ggcorrplot)
# library(gplots)
# library(pheatmap)
# library(EnsDb.Mmusculus.v79)
# library(Rsamtools)
# library(VennDiagram)
# library(biomaRt)
# library(biobtreeR)
# library(memes)
# library(magrittr)
# 

# library(Motif2Site)
# library(updater)
# 
# library(BiocManager)
# library(ChIPQC)
# library(rtracklayer)

# library(trackViewer)
# library(Gviz)

# library(GenomicFeatures)
# library(UpSetR)
# library(grid)
# library(plyr)
# library(tidyGenomeBrowser)
# library(rentrez)
# library("GenomicFeatures")
# library("Gviz")
# library(GenomicRanges)
# library()


```


## R folders
```{r R_folders, warning=FALSE, echo=F}

if(Sys.info()[["sysname"]]== "Linux") {
  s <- "/mnt/s"} else(
  s <- "S:")

ddir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF",sep="/")
gitdir <- paste(ddir,"ChIPseq_Wt1_P3026", sep="/")
resdir <- paste(gitdir,"Results", sep="/")
pubfigdir <- paste(resdir,"pub_figures",sep="/")

wdir <- "E:/Simon/P3026_ChIP_Seq_epiSVF"

# setwd(gitdir)
getwd()


```

# Unix Prepare System
## Unix Update System
```{bash unix_system, eval=FALSE, echo=F}
##  linux ####
sudo apt update
sudo apt upgrade
conda update conda
conda update --all
sudo mount -t drvfs S: /mnt/s

```

## Unix Folder & Names
```{bash unix_folders, message=FALSE, warning=FALSE, echo=FALSE}
source activate chipseq

run="Kelly";

dirdata="/mnt/s/AG/AG-Scholz-NGS/Daten/ChIP_epiSVF_P3026";
dir="/mnt/e/Simon/P3026_ChIP_Seq_epiSVF"; 
dirs="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF"; 

index_GRCh38="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Human/GRCh38_noalt_decoy_as/GRCh38_noalt_decoy_as";
index_GRCm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/GRCm39";
index_mm10="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/mm10/mm10";
index_mm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/USCS/mm39";
index_cm_mm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/USCS/mm39.index";

fastqdir="$dirdata/fastq";

ardir="$dirdata/Adapterremoval";
  test -d $ardir && echo "# Folder $ardir exists" ||
  (mkdir $ardir && echo "# Folder $ardir created");
  
fastqcdir="$dirdata/fastqc/";
  test -d $fastqcdir && echo "# Folder $fastqcdir exists" ||
  (mkdir $fastqcdir && echo "# Folder $fastqcdir created");
  
multiqcdir="$dirdata/multiqc/";
  test -d $multiqcdir && echo "# Folder $multiqcdir exists" ||
  (mkdir $multiqcdir && echo "# Folder $multiqcdir created");
  
bdir="$dir/Bowtie2";
  test -d $bdir && echo "# Folder $bdir exists" ||
  (mkdir $bdir && echo "# Folder $bdir created");
  
mdir="$dirs/MACS3";
  test -d $mdir && echo "# Folder $mdir exists" ||
  (mkdir $mdir && echo "# Folder $mdir created");

rgtmdir="$dir/RGT-Motif";
  test -d $rgtmdir && echo "# Folder $rgtmdir exists" ||
  (mkdir $rgtmdir && echo "# Folder $rgtmdir created");

dtoolsdir="$dir/deeptools";
  test -d $dtoolsdir && echo "# Folder $dtoolsdir exists" ||
  (mkdir $dtoolsdir && echo "# Folder $dtoolsdir created");

chromapdir="$dir/chromap";
  test -d $chromapdir && echo "# Folder $chromapdir exists" ||
  (mkdir $chromapdir && echo "# Folder $chromapdir created");

blacklhs="/mnt/e/Simon/ChIPSeq_Kelly/Bowtie2/hg38-blacklist.v2.bed"
blacklmm39="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/mm39.excluderanges.bed.gz"
blacklmm10="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/mm10/mm10.blacklist.bed.gz"

```

# 3 Results

![Analysis Overview](../Data/sheme.pdf){height=100%, width=100%}

![Analysis Overview](../Data/sheme.png){height=100%, width=100%}


```{r method_overview2, echo = TRUE, include=TRUE, results='markup'}
knitr::include_graphics("https://github.com/DNAborn/ChIPseq_Wt1/blob/main/Results/sheme.png")
knitr::include_graphics("https://github.com/DNAborn/ChIPseq_Wt1/blob/main/Results/sheme.pdf")
# knitr::include_graphics("./sheme.png")
# knitr::include_graphics("./sheme.pdf")
# knitr::include_graphics("../sheme.png")
# knitr::include_graphics("../sheme.pdf")
# knitr::include_graphics("/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/ChIPseq_Wt1_P3026/Results/sheme.png")
# knitr::include_graphics("/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/ChIPseq_Wt1_P3026/Results/sheme.pdf")
getwd()
```


```{r Method_overview, message=TRUE, fig.align='center', fig.cap='Analysis Overview', out.width='1\\linewidth', fig.pos='H'}
print("include figure in r chunk")
knitr::include_graphics("../Data/sheme.pdf") # doesn't show

```


### Generate combined peak list
```{r combine_peaks, echo=FALSE, }
csdir <- paste(ddir,"ChIPSeeker",sep="/")

# Peak Calling files
macsdir <- paste(ddir,"MACS3",sep="/")
macssubdir <- list.dirs(macsdir)
macssubdir <- macssubdir[c(6,5,3,2,4)]

# blacklist
blacklmm39 <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Mus/GRCm39/mm39.excluderanges.bed.gz",sep="/")
bl <- readPeakFile(blacklmm39)

extraCols_narrowPeak <- c(signalValue = "numeric", pValue = "numeric",
                          qValue = "numeric", peak = "integer")

# Read peak files from all subfolders
# generate list with combined peaks
npeak_list_all <- list()
npeak_combined_all <- list()
npeak_numbers_table <- {}
hist_hits <- list()

# loop through subfolders
for (beddir in macssubdir){
  print(beddir)
  run <- basename(beddir)
  print(run)

beds <- list.files(beddir, pattern=".narrowPeak")
beds <- file.path(beddir,beds)
name <- paste(run,"_",sep="")


npeak_list <- list()


i <- beds[1]
# inside loop thourgh each replicate
for (i in beds) {
  npeak_numbers <- {}
  # print(i)
  # npeak <- readPeakFile(i)
npeak <- import(i, format = "BED",extraCols = extraCols_narrowPeak)
n <- basename(i) %>% str_remove(pattern="_all_peaks.narrowPeak")
print(paste("Run: ",run,", Sample: ",n," Peaks: ",length(npeak), sep=""))
npeak_numbers <- rbind(npeak_numbers,data.frame(sample = paste(n,sep="_"), Mapping = run, all_peaks = length(npeak)))
assign(paste(run,"npeaks",sep="_"),npeak_numbers)

print(c("Total peaks",length(npeak)))
npeak <- keepStandardChromosomes(npeak, pruning.mode="coarse")
print(c("Peaks on Std Chromosomes",length(npeak)))
npeak_numbers <- cbind(npeak_numbers, filt_Std_Chrom = length(npeak))
filtered <- GenomicRanges::setdiff(npeak, bl)
revmap <- findOverlaps(filtered, npeak, select="arbitrary")
mcols(filtered) <- mcols(npeak)[revmap, , drop=FALSE]
print(c("Peaks filtered from blacklist:",length(filtered)))
npeak_numbers <- cbind(npeak_numbers, filt_bl = length(filtered))
npeak_list[[n]] <- filtered
npeak_list_all[[n]] <- filtered
npeak_numbers_table <- rbind(npeak_numbers_table,npeak_numbers)
}

# combine replicates
assign(paste(run,"npeak_list",sep="_"),npeak_list)
npeak_list <- GRangesList(npeak_list)
ul <- unlist(npeak_list)
ul <- sort(ul,by=~qValue, decreasing = T)
length(ul)
npeak_combined <- GenomicRanges::reduce(ul)
length(ul)
length(npeak_combined)
revmap <- findOverlaps(npeak_combined, ul, select="first")
mcols(npeak_combined) <- mcols(ul)[revmap, , drop=FALSE]
length(npeak_combined)

# count overlap between peaks
npeak_combined$hits <- rep(0,length(npeak_combined))
for (i in 1:length(npeak_list)){
# print(i)
npeak_combined$hits <- npeak_combined$hits+countOverlaps(npeak_combined,npeak_list[i])
}
# hist(c(0,10,npeak_combined$hits), xlim=c(0,10), breaks=10, main=c("hits per replicate, Method: ",run))

# generate list of histograms


gg_b <- ggplot_build(
  ggplot(as.data.frame(npeak_combined),aes(x=hits, colours=hits)) +
    stat_bin(alpha=0.5, position = 'identity', binwidth = 1, colours = npeak_combined$hits) + 
    scale_fill_viridis_c() +
    coord_cartesian(xlim = c(0, 10)) +
    stat_bin(position = 'identity', binwidth=1, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
    scale_x_continuous(n.breaks= 10) +
    ggtitle(paste("peaks per replicate. Total peaks (>1 hit):"))
)
nu_bins <- dim(gg_b$data[[1]])[1]
ymax <- max(gg_b$data[[1]]$y)
g <- ggplot(as.data.frame(npeak_combined),aes(x=hits, colours=hits)) +
    stat_bin(alpha=0.5, position = 'identity', binwidth = 1, colours = npeak_combined$hits, fill = viridis(nu_bins)) + 
    scale_fill_viridis_c() +
    coord_cartesian(xlim = c(0, 10), ylim=c(0,ymax*1.05)) +
    stat_bin(position = 'identity', binwidth=1, geom='text', size=3,  vjust=-0.2, aes(label=after_stat(count)), colour=c("red",rep("blue",nu_bins-1))) +
    scale_x_continuous(n.breaks= 10) +
    ggtitle(paste("Method:",run,"Peaks per replicate. Total peaks (>1 hit):",summary(elementMetadata(npeak_combined)$hits > 1)["TRUE"]))
# 
print(paste(run,": generate ggplot"))
hist_hits[[run]] <- g

summary(elementMetadata(npeak_combined)$hits > 1)[2]
npeak_combined <- npeak_combined[elementMetadata(npeak_combined)$hits > 1,]
length(npeak_combined)
assign(paste(run,"npeak_combined",sep="_"),npeak_combined)

npeak_combined_all[[run]] <- npeak_combined

}

# npeak_combined_all
# lapply(npeak_combined_all, length)

# generate overlap table
peaks_overlap <- data.frame(Sample=basename(macssubdir),
                          Peaks=lengths(npeak_combined_all))
ol.all <- {}
l <- length(npeak_combined_all)
for (n in names(npeak_combined_all)){
  assign(paste("ol",n,sep="."),{})
  variable <- paste0("ol.",n)
  ol <- eval(parse(text = variable))
  peaks_to_compare <- npeak_combined_all[[n]]
for (i in 1:l){
  start(peaks_to_compare) <- start(peaks_to_compare) - 100
  end(peaks_to_compare) <- end(peaks_to_compare) + 100
  ol <- c(ol,sum(countOverlaps(peaks_to_compare,npeak_combined_all[[i]])))
}
  assign(paste("ol",n,sep="."),ol)
}

peaks_overlap <- data.frame(peaks_overlap,ol.Std, ol.pe, ol.local, ol.chro5k, ol.Online)
# peaks_overlap

```

#### Peak Tables
```{r peak_numbers, include=FALSE}
knitr::kable(npeak_numbers_table) %>% 
  kable_styling("striped", full_width = F) %>% 
 scroll_box(height = "400px")
# datatable(npeak_numbers_table)
knitr::kable(peaks_overlap)
# tibble::as.tibble(npeak_numbers_table)
# tibble::as.tibble(peaks_overlap)
```

#### Hists & Tables
```{r hits_tables, out.width="100%", echo=FALSE, out.height="2000px"}

# patchwork::wrap_plots(hist_hits) +plot_layout(nrow = 3, ncol = 2, axis_titles = "collect") # axes = "collect"

# add sum row to tables
ht_list <- list()
for (run in names(hist_hits)){
     table <- subset(npeak_numbers_table, Mapping == run) %>% 
                bind_rows(summarise_all(., ~if(is.numeric(.)) sum(.) else "Total"))
      l <- dim(table)[1]
      t1 <- ttheme_default(base_size = 8,
                         core=list(
        fg_params=list(fontface=c(rep("plain", (l-1)), "bold.italic")),
        bg_params = list(fill=c(rep(c("white", "grey95"),
                                    length.out=(l-1)), "lightyellow")
        )))
      
  ht_list[[run]] <-  hist_hits[[run]] / gridExtra::tableGrob(table, theme = t1) +  plot_layout(heights = c(1,2))
}
patchwork::wrap_plots(ht_list[1:2]) + plot_layout(ncol = 2)
patchwork::wrap_plots(ht_list[3:4]) + plot_layout(ncol = 2)
patchwork::wrap_plots(ht_list[5]) + plot_layout(ncol = 2)

#  bg_params = list(fill=c(rep("white",(l-1)), "skyblue1"))) 

```




#### Overlap Peaks
```{r venn_overlaps,  echo = FALSE}

# Venns
olvenn <- findOverlapsOfPeaks(npeak_combined_all[[1]],
                          npeak_combined_all[[2]],
                          npeak_combined_all[[3]],
                          npeak_combined_all[[4]],
                          npeak_combined_all[[5]])

length(npeak_combined_all)
# pdf(paste(csrundir,"Venn_Peaks2.pdf",sep="/"))
makeVennDiagram(main="Overlap of combined & filtered (>1 hits) peak lists",
                plot = TRUE, npeak_combined_all,
                NameOfPeaks = names(npeak_combined_all),
                fill=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
                cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise")
                )

# dev.off()
# ggsave(peaks.venn,file=paste(pubfigdir,"Venn_Peaks.pdf",sep="/"), width = 8, height = 8, device = "pdf")


```

## Annotate Peaks
#### 1 Run
```{r annotated_peaks, include=TRUE, results = "show"}

# Add Gene names
ah = AnnotationHub()
query(ah, c("EnsDb", "Musculus"))
edb <- ah[["AH113713"]]

# 1 Subfolder

figures_annotated_peaks <- list()

i <- "pe"
all_npeaksAnno <- annotatePeak(npeak_combined_all[[i]], TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
  all_npeaksAnno_table <- as.data.frame(all_npeaksAnno)
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]

# Annotated table
top_hits <- subset(all_npeaksAnno_table, distanceToTSS > -2000 & distanceToTSS < 2000) 
dim(top_hits)
top_hits <- top_hits[order(top_hits$score, decreasing=T),]
knitr::kable(top_hits[c(0:50),c("symbol","annotation","distanceToTSS","peak","score","signalValue","qValue","hits")],format = "markdown")

hits_lim <- max(all_npeaksAnno_table$hits)
score_lim <- quantile(all_npeaksAnno_table$score, .90)
peak_lim <- quantile(all_npeaksAnno_table$peak, .90)
signalValue_lim <- quantile(all_npeaksAnno_table$signalValue, .90)
qValue_lim <- quantile(all_npeaksAnno_table$qValue, .90)


g1 <- ggplot(all_npeaksAnno_table,aes(x=score, y=peak, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(0,peak_lim)) +
  ggtitle(paste("score to peak"))

g2 <- ggplot(all_npeaksAnno_table,aes(x=score, y=signalValue, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(0,signalValue_lim)) +
  ggtitle(paste("score to signalValue"))

g3 <- ggplot(all_npeaksAnno_table,aes(x=score, y=qValue, color=hits)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c() +
  coord_cartesian(xlim=c(0,score_lim), ylim = c(0,qValue_lim)) +
  ggtitle(paste("score to qValue"))

g1+g2+g3+plot_layout(guides = "collect")


head(all_npeaksAnno_table)

ggplot(all_npeaksAnno_table,aes(x=distanceToTSS, y=score, color=hits)) +
  geom_point(size=1, alpha=0.5, position = "jitter") +
  scale_color_viridis_c() +
  coord_cartesian(ylim=c(0,score_lim)) +
  ggtitle(paste("scores to signalValue"))
# xlim=c(-10000,10000)



for (i in 1:length(npeak_combined_all)){
all_npeaksAnno <- annotatePeak(npeak_combined_all[[i]], TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
  all_npeaksAnno_table <- as.data.frame(all_npeaksAnno)
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]

g1 <- ggplot(all_npeaksAnno_table,aes(x=hits)) + # fill = cut(hits, 100)
      stat_bin(alpha=0.6, position = 'identity', binwidth=1) + 
      scale_fill_viridis_d() +
      stat_bin(position = 'identity', binwidth=1, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
      ggtitle("peaks in replicates")

g2a <- ggplot(all_npeaksAnno_table,aes(x=score, group=annotation_short, fill=annotation_short)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=5) + 
  # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
  scale_fill_viridis_d() +
  ggtitle(paste("genetic region per score (",n,")",sep="")) + coord_cartesian(xlim = c(0, 100))
g2b <- ggplot(all_npeaksAnno_table,aes(x=signalValue, group=annotation_short, fill=annotation_short)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=1) + 
  # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
  scale_fill_viridis_d() +
  ggtitle(paste("genetic region per signalValue (",n,")",sep="")) + coord_cartesian(xlim = c(0, 15))

g3 <- ggplot(all_npeaksAnno_table,aes(x=width, group=hits, fill=hits)) +
      stat_bin(alpha=0.5, position = 'identity', binwidth=200) + 
  # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
  scale_fill_viridis_c() +
  ggtitle(paste("hits per peak size")) + coord_cartesian(xlim = c(0, 10000))

all_npeaksAnno_table$hits <- as.factor(all_npeaksAnno_table$hits)

# g4 <- ggplot(all_npeaksAnno_table,aes(x=hits, y=qValue)) +
#   geom_violin(aes(fill=hits)) +
#   scale_fill_viridis_d() +
#   geom_point(size=0.05, position ='jitter', alpha=0.2,color="grey40") +
#   coord_cartesian(ylim=c(0,50))

g4 <- ggplot(all_npeaksAnno_table,aes(x=hits, y=score)) +
  geom_violin(aes(fill=hits, color=hits)) +
  scale_fill_viridis_d() +
  geom_point(size=0.05, position ='jitter', alpha=0.2,color="grey40") +
  coord_cartesian(ylim=c(0,1000)) +
  ggtitle(paste("scores per hit"))

levels(factor(all_npeaksAnno_table$annotation_short))

g5 <- ggplot(all_npeaksAnno_table,aes(x=score, y=qValue, color=signalValue)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=0.5, position ='jitter', alpha=0.5) +
  scale_color_viridis_c(limits = c(0, 15)) +
  coord_cartesian(ylim=c(0,50), xlim = c(0,500)) +
  ggtitle(paste("score to qvalue"))

g6 <- ggplot(all_npeaksAnno_table,aes(x=score, y=signalValue, color=pValue)) +
  # geom_violin(aes(fill=hits)) +
  geom_point(size=1, alpha=0.5) +
  scale_color_viridis_c(option = "D", direction = -1, limits = c(0, 100)) +
  coord_cartesian(ylim=c(0,15), xlim = c(0,1000))+
  ggtitle(paste("scores to signalValue"))

# datatable(head(all_npeaksAnno_table[order(all_npeaksAnno_table$width),]))
viridis(8)

top_hits <- subset(all_npeaksAnno_table, distanceToTSS > -2000 & distanceToTSS < 2000) 
dim(top_hits)
top_hits <- top_hits[order(top_hits$score, decreasing=T),]
top_hits_list[[n]] <- top_hits

figures_annotated_peaks[[n]] <- g1+g2a+g3+g4+g5+g6+plot_layout(nrow = 2, ncol = 3, axis_titles = "collect") + plot_annotation(title = paste("Method:",n))
}



options(kableExtra.auto_format = FALSE)
knitr::kable(top_hits[c(0:50),c("symbol","distanceToTSS","hits","score","signalValue","qValue")],format = "markdown")
# datatable(top_hits[c(0:50),c("symbol","distanceToTSS","hits","score","signalValue","qValue")])

```

#### All peaks
```{r eval=FALSE, include=FALSE}

## All peaks
  
npeakAnnoList <- lapply(npeak_combined_all, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)

# make list
npeakAnnoList_table <- list()
gene_ids_all_list <- list()
for (i in 1:length(npeakAnnoList)){
  print(paste(i,": ",names(npeakAnnoList)[i],sep=""))
  n <- names(npeakAnnoList[i])
  table <- as.data.frame(npeakAnnoList[[i]]@anno)
  table$transcriptId2  <- sub("\\.\\d+$", "", table$transcriptId)
  table$entrez <- table$geneId
  table$geneId <- mapIds(edb, keys = table$transcriptId2, column = "GENEID", keytype = "TXID")
  table$symbol <- mapIds(edb, keys = table$transcriptId2, column = "SYMBOL", keytype = "TXID")
  assign(paste("narrow_table",n,sep="_"),table)
  table_filter <- subset(table, distanceToTSS > -2000 & distanceToTSS < 2000)
  gene_ids_all_list <- lapply(gene_ids_all_list,unique)

  print(paste("Before: ",length(rownames(table)),", After: ",length(rownames(table_filter))," (",round(length(rownames(table_filter))/length(rownames(table))*100),"%)"))
  npeakAnnoList_table[[n]] <- table_filter
  gene_ids_all_list[[n]] <- table_filter$geneId
}

# Results from RNA-Seq
deg <- readRDS("../../RNA-Seq_res_degs.RDS")
deg.top <- readRDS("../../RNA-Seq_top_degs.RDS")

# Plot gene region for all lists
i <- 5
for (i in names(npeakAnnoList)){

  all_npeaksAnno_table <- as.data.frame(npeakAnnoList[[i]])
  all_npeaksAnno_table$transcriptId2 <- sub("\\.\\d+$", "", all_npeaksAnno_table$transcriptId)
  all_npeaksAnno_table$geneId <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "GENEID", keytype = "TXID")
  all_npeaksAnno_table$symbol <- mapIds(edb, keys = all_npeaksAnno_table$transcriptId2, column = "SYMBOL", keytype = "TXID")
n <- names(npeak_combined_all[i])
all_npeaksAnno_table$annotation_short <-  str_split(all_npeaksAnno_table$annotation,pattern = " ", simplify = TRUE)[,1]

g <- ggplot(all_npeaksAnno_table,aes(x=symbol, group=annotation_short, fill=annotation_short)) +
      stat_bin(alpha=0.5, position = 'identity') + 
  # stat_bin(position = 'identity', binwidth=100, geom="text", aes(label=after_stat(count)), vjust=-0.5, colour="blue") +
  scale_fill_viridis_d() +
  ggtitle(paste("genetic region per peak size (",n,")",sep="")) # + coord_cartesian(xlim = c(0, 100), ylim=c(0,200))
plot(g)
# ggsave(g, file=paste(pubfigdir,"/Peaks_geneticregions_score_",n,".pdf",sep=""), width = 8, height = 8, device = "pdf")
}


```

#### Venns
```{r eval=FALSE, include=FALSE}
# overlap
gene_ids_all_list[["RNASeq"]] <- deg.top$gene_id

id_comb <- unlist(gene_ids_all_list) %>% unique()
id_symb <- mapIds(edb, keys = id_comb, column = "SYMBOL", keytype = "GENEID", multiVals = "first")

id_com_table <- data.frame("ensembl" = id_comb,
                           "id_symbol" = id_symb) 

for (n in names(gene_ids_all_list)){
  isin <- {}
  for (i in id_com_table$ensembl){
    ifelse(i %in% gene_ids_all_list[[n]],
         isin <- c(isin,1),
         isin <- c(isin,0))
  }
  id_com_table <- cbind(id_com_table,isin)
  l <- ncol(id_com_table)
  colnames(id_com_table)[l] <- n
}
id_com_table

subset(id_com_table, chro5k == 1)
subset(id_com_table, Std == 1)
subset(id_com_table, pe == 1 & local == 1 & Online == 0 & RNASeq == 1)$id_symbol

# & Online == TRUE
# & RNASeq == TRUE

id_com_table$sum <- rowSums(id_com_table[,3:8])
summary(duplicated(id_com_table$id_symbol))
id_com_table <- id_com_table[!duplicated(id_com_table$id_symbol),]
id_com_table <- id_com_table[!is.na(id_com_table$id_symbol),]
id_com_table <- id_com_table[order(id_com_table$sum, decreasing = TRUE),]
rownames(id_com_table) <- id_com_table$id_symbol

heatmap(Rowv = TRUE, as.matrix(subset(id_com_table,sum>1)[,3:8]))

pdf(file=paste(csdir,"/Heatmap.pdf",sep=""))
heatmap(as.matrix(subset(id_com_table,Std == 1 | chro5k == 1 | pe == 1 & local ==1)[,3:8]),
        Rowv = TRUE)
heatmap(as.matrix(subset(id_com_table,Std == 1 | chro5k == 1 | pe == 1 | local ==1)[,3:8]),
        Rowv = FALSE)
dev.off()
ggsave(file=paste(pubfigdir,"/Heatmap.png",sep=""), width = 8, height = 8)

heat.data <- subset(id_com_table,Std == 1 | chro5k == 1 | pe == 1 & local ==1)[,3:8]
ggplot(heat.data, aes(x=c(colnames(heat.data)),y=rownmaes(), fill= Z)) + 
  geom_tile()

lapply(gene_ids_all_list,length)
gene_ids_all_list <- lapply(gene_ids_all_list,unique)
lapply(gene_ids_all_list,length)

length(gene_ids_all_list)
vennplot(gene_ids_all_list[1:5])
ggvenn(gene_ids_all_list[1:4],
      fill_color = c("yellow3", "yellow", "yellow","darkolivegreen1"),
      stroke_size = 0.5,
      show_percentage = FALSE)
ggsave(file=paste(pubfigdir,"/Venn_Genes_own.png",sep=""), width = 8, height = 8)

in_pe_local <- intersect(gene_ids_all_list[["pe"]],
          gene_ids_all_list[["local"]])
in_pe_local_Std <- intersect(in_pe_local,gene_ids_all_list[["Std"]])
in_pe_local_Std

,
          gene_ids_all_list[["pe"]]
,
          gene_ids_all_list[["local"]],
          gene_ids_all_list[["Online"]],
          gene_ids_all_list[["chro5k"]],

)


setdiff (intersect(intersect(gene_ids_all_list["chro5k"], gene_ids_all_list["Std"]), UPDEGsTR40SR40genes),  UPDEGsTRCSRCgenes)


ggvenn(gene_ids_all_list[c(2,3,5,6)],
      fill_color = c("yellow", "yellow","lightblue","salmon"),
      stroke_size = 0.5,
      show_percentage = FALSE)
ggsave(file=paste(pubfigdir,"/Venn_Genes_compare.png",sep=""), width = 8, height = 8)

ol <- calculate.overlap(gene_ids_all_list[1:5])


       

library(ggplot2)
library(ggVennDiagram)

x <- list(A=1:5,B=2:7,C=3:6,D=4:9)




# manual
venn <- Venn(x)
data <- process_data(venn)
ggplot() +
  # 1. region count layer
  geom_sf(aes(fill = count), data = venn_region(data)) +
  # 2. set edge layer
  geom_sf(aes(color = name), data = venn_setedge(data), show.legend = TRUE, size = 2) +
  # 3. set label layer
  geom_sf_text(aes(label = name), data = venn_setlabel(data)) +
  # 4. region label layer
  geom_sf_label(aes(label = paste0(count, " (", scales::percent(count/sum(count), accuracy = 2), ")")), 
                data = venn_region(data),
                size = 3) +
  scale_fill_gradient(low = "#F4FAFE", high = "#4981BF")+
  scale_color_manual(values = c("A" = "yellow","B" ="steelblue",'C' = 'red', 'D' = 'black'),
                     labels = c('D' = 'D = bdiv_human'))+
  theme_void()




v <- ggVennDiagram(gene_ids_all_list,
              label_alpha = 0,
              label = "count",
              set_color = c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"))
v + scale_fill_distiller(palette = "RdBu")

              ,
              
              scale_fill_distiller(palette = "RdBu"))
,
              scale_fill_manual = c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
              cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"))
              )

makeVennDiagram(gene_ids_all_list)

,by=gplots)

                         fill=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise"),
                cat.col=c("#009E73", "lightgreen", "lightgreen","lightblue","turquoise")
                )
       
       
ol <- calculate.overlap(gene_ids_all_list[1:3])
summary(ol)
ol <- calculate.overlap(list("ChIP.all"=subset(gene_table_hits, hits > 7)$geneid,
                             "RNA-Seq"=deg.top$gene_id))



venn.diagram(gene_ids_all_list)

chip.rna <- sort(unname(mapIds(edb,keys=ol$a3,column = "SYMBOL",keytype = "GENEID", multiVals = "first")))

beep()


```

