<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kelterborn" />

<meta name="date" content="2024-01-31" />

<title>ChIP_Wt1 - Results</title>

<script src="README_files/header-attrs-2.25/header-attrs.js"></script>
<script src="README_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="README_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="README_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="README_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="README_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="README_files/navigation-1.1/tabsets.js"></script>
<link href="README_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="README_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">ChIP_Wt1 - Results</h1>
<h4 class="author">Kelterborn</h4>
<h4 class="date">2024-01-31</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#r-prepare-system" id="toc-r-prepare-system">R Prepare
System</a>
<ul>
<li><a href="#r-update-and-load-librarys"
id="toc-r-update-and-load-librarys">R update and load librarys</a></li>
<li><a href="#r-folders" id="toc-r-folders">R folders</a></li>
</ul></li>
<li><a href="#unix-prepare-system" id="toc-unix-prepare-system">Unix
Prepare System</a>
<ul>
<li><a href="#unix-update-system" id="toc-unix-update-system">Unix
Update System</a></li>
<li><a href="#unix-folder-names" id="toc-unix-folder-names">Unix Folder
&amp; Names</a></li>
</ul></li>
<li><a href="#results" id="toc-results">3 Results</a>
<ul>
<li><a href="#generate-combined-peak-list"
id="toc-generate-combined-peak-list">Generate combined peak
list</a></li>
<li><a href="#annotate-peaks" id="toc-annotate-peaks">Annotate
Peaks</a></li>
</ul></li>
</ul>
</div>

<!-- <style> -->
<!-- .vscroll-plot { -->
<!--     width: 1000px; -->
<!--     height: 1000px; -->
<!--     overflow-y: scroll; -->
<!--     overflow-x: scroll; -->
<!-- } -->
<!-- </style> -->
<div id="r-prepare-system" class="section level1">
<h1>R Prepare System</h1>
<div id="r-update-and-load-librarys" class="section level2">
<h2>R update and load librarys</h2>
<p>BiocManager::install(““)</p>
<pre class="r"><code>BiocManager::install(update = TRUE, ask = FALSE)
library(dbplyr)
library(tidyverse)
library(ChIPseeker)
library(rtracklayer)
library(trackViewer)
library(GenomicRanges)
library(IRanges)
library(ChIPpeakAnno)
library(AnnotationHub)
library(ggplot2)
library(viridis)
library(kableExtra)
library(DT)
library(patchwork)
library(gridExtra)

library(TxDb.Mmusculus.UCSC.mm39.knownGene)
txdb &lt;- TxDb.Mmusculus.UCSC.mm39.knownGene</code></pre>
</div>
<div id="r-folders" class="section level2">
<h2>R folders</h2>
</div>
</div>
<div id="unix-prepare-system" class="section level1">
<h1>Unix Prepare System</h1>
<div id="unix-update-system" class="section level2">
<h2>Unix Update System</h2>
</div>
<div id="unix-folder-names" class="section level2">
<h2>Unix Folder &amp; Names</h2>
</div>
</div>
<div id="results" class="section level1">
<h1>3 Results</h1>
<div class="float">
<embed src="../Data/sheme.pdf" style="width:100.0%" />
<div class="figcaption">Analysis Overview</div>
</div>
<div class="float">
<img src="../Data/sheme.png" style="width:100.0%"
alt="Analysis Overview" />
<div class="figcaption">Analysis Overview</div>
</div>
<pre class="r"><code>knitr::include_graphics(&quot;https://github.com/DNAborn/ChIPseq_Wt1/blob/main/Results/sheme.png&quot;)
knitr::include_graphics(&quot;https://github.com/DNAborn/ChIPseq_Wt1/blob/main/Results/sheme.pdf&quot;)
# knitr::include_graphics(&quot;./sheme.png&quot;)
# knitr::include_graphics(&quot;./sheme.pdf&quot;)
# knitr::include_graphics(&quot;../sheme.png&quot;)
# knitr::include_graphics(&quot;../sheme.pdf&quot;)
# knitr::include_graphics(&quot;/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/ChIPseq_Wt1_P3026/Results/sheme.png&quot;)
# knitr::include_graphics(&quot;/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/P3026_ChIP-Seq_epiSVF/ChIPseq_Wt1_P3026/Results/sheme.pdf&quot;)
getwd()</code></pre>
<pre class="r"><code>print(&quot;include figure in r chunk&quot;)
knitr::include_graphics(&quot;../Data/sheme.pdf&quot;) # doesn&#39;t show</code></pre>
<div id="generate-combined-peak-list" class="section level3">
<h3>Generate combined peak list</h3>
<div id="peak-tables" class="section level4">
<h4>Peak Tables</h4>
</div>
<div id="hists-tables" class="section level4">
<h4>Hists &amp; Tables</h4>
<p><img src="README_files/figure-html/hits_tables-1.png" width="100%" height="1000px" /><img src="README_files/figure-html/hits_tables-2.png" width="100%" height="1000px" /><img src="README_files/figure-html/hits_tables-3.png" width="100%" height="1000px" /><img src="README_files/figure-html/hits_tables-4.png" width="100%" height="1000px" /><img src="README_files/figure-html/hits_tables-5.png" width="100%" height="1000px" /></p>
</div>
<div id="overlap-peaks" class="section level4">
<h4>Overlap Peaks</h4>
<p><img
src="README_files/figure-html/venn_overlaps-1.png" /><!-- --></p>
</div>
</div>
<div id="annotate-peaks" class="section level2">
<h2>Annotate Peaks</h2>
<div id="run" class="section level4">
<h4>1 Run</h4>
<p><img
src="README_files/figure-html/unnamed-chunk-1-1.png" /><!-- --></p>
</div>
<div id="all-peaks" class="section level4">
<h4>All peaks</h4>
</div>
<div id="venns" class="section level4">
<h4>Venns</h4>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
